Model/Ach/Ui/ConfigProvider.php

```diff
@@ -0,0 +1,130 @@
+{
+    const METHOD_CODE = 'braintree_ach_direct_debit';
+
+    const CONFIG_MERCHANT_COUNTRY = 'paypal/general/merchant_country';
+
+    const CONFIG_STORE_NAME = 'general/store_information/name';
+
+    const ALLOWED_MERCHANT_COUNTRIES = ['US'];
+
+    /**
+     * @var BraintreeAdapter $adapter
+     */
+    private $adapter;
+    /**
+     * @var BraintreeConfig $braintreeConfig
+     */
+    private $braintreeConfig;
+    /**
+     * @var ScopeConfigInterface $scopeConfig
+     */
+    private $scopeConfig;
+    /**
+     * @var string $clientToken
+     */
+    private $clientToken = '';
+
+    /**
+     * ConfigProvider constructor.
+     *
+     * @param BraintreeAdapter $adapter
+     * @param BraintreeConfig $braintreeConfig
+     * @param ScopeConfigInterface $scopeConfig
+     */
+    public function __construct(
+        BraintreeAdapter $adapter,
+        BraintreeConfig $braintreeConfig,
+        ScopeConfigInterface $scopeConfig
+    ) {
+        $this->adapter = $adapter;
+        $this->braintreeConfig = $braintreeConfig;
+        $this->scopeConfig = $scopeConfig;
+    }
+
+    /**
+     * Retrieve assoc array of checkout configuration
+     *
+     * @return array
+     * @throws InputException
+     * @throws NoSuchEntityException
+     */
+    public function getConfig(): array
+    {
+        return [
+            'payment' => [
+                self::METHOD_CODE => [
+                    'isActive' => $this->isActive(),
+                    'clientToken' => $this->getClientToken(),
+                    'storeName' => $this->getStoreName()
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * ACH is for the US only.
+     * Logic based on Merchant Country Location config option.
+     *
+     * @return bool
+     */
+    public function isActive(): bool
+    {
+        $merchantCountry = $this->scopeConfig->getValue(
+            self::CONFIG_MERCHANT_COUNTRY,
+            ScopeInterface::SCOPE_STORE
+        );
+
+        return in_array($merchantCountry, self::ALLOWED_MERCHANT_COUNTRIES, true);
+    }
+
+    /**
+     * @return string
+     * @throws InputException
+     * @throws NoSuchEntityException
+     */
+    public function getClientToken(): string
+    {
+        if (empty($this->clientToken)) {
+            $params = [];
+
+            $merchantAccountId = $this->braintreeConfig->getMerchantAccountId();
+            if (!empty($merchantAccountId)) {
+                $params[PaymentDataBuilder::MERCHANT_ACCOUNT_ID] = $merchantAccountId;
+            }
+
+            $this->clientToken = $this->adapter->generate($params);
+        }
+
+        return $this->clientToken;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getStoreName()
+    {
+        $storeName = $this->scopeConfig->getValue(
+            self::CONFIG_STORE_NAME,
+            ScopeInterface::SCOPE_STORE
+        );
+        return $storeName ?: 'this store';
```

Comment:
this store?

Reply:
So if no store name is set, I need some form of basic placeholder name to go into the mandate text string. Wasn't sure what else to put :D

Reply:
you could use this:
```
return $this->_scopeConfig->getValue(
        'general/store_information/name',
        \Magento\Store\Model\ScopeInterface::SCOPE_STORE
    );
```
will return the store name

Reply:
That's what I'm doing, but sometimes it's empty :)

Reply:
mh, let me think if there is another way

Reply:
alternatively you could take the email sender name (but that could be empty too) or even better the domain name as last resort
i am just thinking to make it more dynamic as possible

Reply:
Ooh, the domain could be a shout! That is practically a certain that is has a value, correct?

Reply:
yes correct! that's need to be set ;)